HistoryManager/src/main/java/eu/neclab/ngsildbroker/historymanager/service/HistoryService.java

```diff
@@ -0,0 +1,177 @@
+import eu.neclab.ngsildbroker.commons.datatypes.requests.BaseRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.CreateHistoryEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.DeleteEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.DeleteHistoryEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.HistoryEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.requests.UpdateHistoryEntityRequest;
+import eu.neclab.ngsildbroker.commons.datatypes.results.QueryResult;
+import eu.neclab.ngsildbroker.commons.datatypes.results.UpdateResult;
+import eu.neclab.ngsildbroker.commons.enums.ErrorType;
+import eu.neclab.ngsildbroker.commons.exceptions.ResponseException;
+import eu.neclab.ngsildbroker.commons.interfaces.EntryCRUDService;
+import eu.neclab.ngsildbroker.commons.ngsiqueries.ParamsResolver;
+import eu.neclab.ngsildbroker.commons.querybase.BaseQueryService;
+import eu.neclab.ngsildbroker.commons.storage.StorageDAO;
+import eu.neclab.ngsildbroker.commons.tools.HttpUtils;
+import eu.neclab.ngsildbroker.historymanager.repository.HistoryDAO;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.groups.UniAndGroup2;
+import io.smallrye.mutiny.unchecked.Unchecked;
+import io.smallrye.reactive.messaging.MutinyEmitter;
+
+@Singleton
+public class HistoryService extends BaseQueryService implements EntryCRUDService {
+
+	private final static Logger logger = LoggerFactory.getLogger(HistoryService.class);
+
+	@Inject
+	HistoryDAO historyDAO;
+
+	@ConfigProperty(name = "scorpio.directDB", defaultValue = "true")
+	boolean directDB;
+
+	@ConfigProperty(name = "scorpio.topics.temporal")
+	String TEMP_TOPIC;
+
+	@ConfigProperty(name = "scorpio.history.tokafka", defaultValue = "false")
+	boolean historyToKafkaEnabled;
+
+	@Inject
+	@Channel(AppConstants.HISTORY_CHANNEL)
+	MutinyEmitter<BaseRequest> kafkaSenderInterface;
+
+	public Uni<String> createEntry(ArrayListMultimap<String, String> headers, Map<String, Object> resolved) {
+		return createTemporalEntity(headers, resolved, false);
+	}
+
+	Uni<String> createTemporalEntity(ArrayListMultimap<String, String> headers, Map<String, Object> resolved,
+			boolean fromEntity) {
+		logger.trace("creating temporal entity");
+		CreateHistoryEntityRequest request;
+		try {
+			request = new CreateHistoryEntityRequest(headers, resolved, fromEntity);
+		} catch (Exception e) {
+			return Uni.createFrom().failure(e);
+		}
+		return handleRequest(request).combinedWith((t, u) -> {
+			logger.debug("createMessage() :: completed");
+			return request.getId();
+		});
+
+	}
+
+	public Uni<Boolean> deleteEntry(ArrayListMultimap<String, String> headers, String entityId) {
+		return delete(headers, entityId, null, null, null);
+	}
+
+	public Uni<Boolean> delete(ArrayListMultimap<String, String> headers, String entityId, String attributeId,
+			String instanceId, Context linkHeaders) {
```

Comment:
Validate Delete functionality if entity is not exist..

Reply:
done